File: Server/runes/SetMana.js
exports.execute = function(rune, state)
{
    var controller = state.controllers[rune.controllerGuid];
    controller.mana = rune.mana;
}

exports.CreateRune = function(controllerGuid, mana)
{
    var rune = {
        "runeType":"SetMana",
        "controllerGuid":controllerGuid,
        "mana":mana
    }
    
    return rune;
}

exports.canSee = function(rune, controller, state)
{
    return true;
}

File: Assets/Scripts/Runes/SetMana.cs
using UnityEngine;
using System.Collections;
using System;

public class SetMana : Rune {

    public string controller_uid { get; set; }
    public int mana { get; set; }

    public override void Execute(Action action)
    {
        Controller playerController = EntityManager.Singelton.GetEntity(controller_uid) as Controller;
        if(playerController == null)
        {
            Debug.Log("Problem finding controller with guid" + controller_uid);
            action();
            return;
        }

        playerController.SetMana(mana);
        action();
    }

    public override void OnGUI()
    {

    }

}
